File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPExhaustion.java
package org.noxet.noxetserver.playerstate.properties;

import org.bukkit.configuration.ConfigurationSection;
import org.bukkit.entity.Player;
import org.noxet.noxetserver.playerstate.PlayerStateProperty;

public class PSPExhaustion implements PlayerStateProperty<Float> {
    @Override
    public String getConfigName() {
        return "exhaustion";
    }

    @Override
    public Float getDefaultSerializedProperty() {
        return 0F;
    }

    @Override
    public Float getSerializedPropertyFromPlayer(Player player) {
        return player.getExhaustion();
    }

    @Override
    public void restoreProperty(Player player, Float exhaustion) {
        player.setExhaustion(exhaustion);
    }

    @Override
    public Float getValueFromConfig(ConfigurationSection config) {
        return ((Double) config.getDouble(getConfigName())).floatValue();
    }
}

File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPExhaustion.scala
package org.noxet.noxetserver.playerstate.properties

import org.bukkit.configuration.ConfigurationSection
import org.bukkit.entity.Player
import org.noxet.noxetserver.playerstate.PlayerStateProperty

object PSPExhaustion extends PlayerStateProperty[Float]:
  override def getConfigName: String = "exhaustion"

  override def getDefaultSerializedProperty: Float = 0F

  override def getSerializedPropertyFromPlayer(player: Player): Float =
    player.getExhaustion

  override def restoreProperty(player: Player, exhaustion: Float): Unit = player.setExhaustion(exhaustion)

  override def getValueFromConfig(config: ConfigurationSection): Float =
    config.getDouble(getConfigName).floatValue